test/serviceprofiles/serviceprofiles_test.go

```diff
@@ -0,0 +1,170 @@
+package serviceprofiles
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+type testCase struct {
+	args           []string
+	deployName     string
+	expectedRoutes []string
+	namespace      string
+	sourceName     string
+	spName         string
+}
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+func TestServiceProfiles(t *testing.T) {
+
+	testNamespace := TestHelper.GetTestNamespace("serviceprofile-test")
+	out, _, err := TestHelper.LinkerdRun("inject", "testdata/tap_application.yaml")
+	if err != nil {
+		t.Fatalf("linkerd inject command failed\n%s", out)
+	}
+
+	out, err = TestHelper.KubectlApply(out, testNamespace)
+	if err != nil {
+		t.Fatalf("kubectl apply command failed\n%s", out)
+	}
+
+	// wait for deployments to start
+	for _, deploy := range []string{"t1", "t2", "t3", "gateway"} {
+		if err := TestHelper.CheckPods(testNamespace, deploy, 1); err != nil {
+			t.Error(err)
+		}
+
+		if err := TestHelper.CheckDeployment(testNamespace, deploy, 1); err != nil {
+			t.Error(fmt.Errorf("Error validating deployment [%s]:\n%s", deploy, err))
+		}
+	}
+
+	testCases := []testCase{
+		{
+			sourceName:     "tap",
+			namespace:      testNamespace,
+			deployName:     "deploy/t1",
+			spName:         "t1-svc",
+			expectedRoutes: []string{"POST /buoyantio.bb.TheService/theFunction", "[DEFAULT]"},
+		},
+		{
+			sourceName:     "open-api",
+			namespace:      testNamespace,
+			spName:         "t3-svc",
+			deployName:     "deploy/t3",
+			expectedRoutes: []string{"DELETE /testpath", "GET /testpath", "PATCH /testpath", "POST /testpath", "[DEFAULT]"},
+		},
```

Comment:
I thought there used to be a test case for `proto`.

Reply:
There was, but as I was going through this I noticed that `TestServiceProfilesDeploy` already exercises this command in `install_test.go`. Do you think it would be better to move that into here?

Reply:
ah.. :eyes: we can keep the existing one in `install_test.go`. Thanks.